packages/flutter/test/material/back_button_test.dart

```diff
@@ -33,6 +34,31 @@ void main() {
     expect(find.text('Home'), findsOneWidget);
   });
 
+  testWidgets('BackButton platform navigation test', (WidgetTester tester) async {
+    final List<MethodCall> log = <MethodCall>[];
+
+    SystemChannels.platform.setMockMethodCallHandler((MethodCall methodCall) async {
+      log.add(methodCall);
+    });
+
+    await tester.pumpWidget(
+        new MaterialApp(
+            home: const Material(
+                child: const Center(
+                  child: const BackButton(),
+                )
+            ),
+        )
+    );
+
+    await tester.tap(find.byType(BackButton));
+
+    await tester.pumpAndSettle();
+
+    expect(find.byType(BackButton), findsOneWidget);
+    expect(log.contains(const MethodCall('SystemNavigator.pop')), true);
```

Comment:
You could also use
```dart
expect(log, contains(const MethodCall('SystemNavigator.pop')));
```
which does exactly the same, but gives a better description if it fails.